catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -307,35 +365,38 @@ private DeleteRequest processPreDeletePolicyPlugins(DeleteRequest deleteRequest,
         return deleteRequest;
     }
 
-    private List<Metacard> populateMetacards(DeleteRequest deleteRequest)
+    private DeleteRequest populateMetacards(DeleteRequest deleteRequest)
             throws IngestException, StopProcessingException {
-        if (frameworkProperties.getPolicyPlugins()
-                .isEmpty()) {
-            return new ArrayList<>();
-        }
+        final String attributeName = deleteRequest.getAttributeName();
 
         QueryRequestImpl queryRequest = createQueryRequest(deleteRequest);
-
+        QueryResponse query;
         try {
-            QueryResponse query = queryOperations.doQuery(queryRequest,
+            query = queryOperations.doQuery(queryRequest,
                     frameworkProperties.getFederationStrategy(),
                     false);
-            List<Metacard> metacards = query.getResults()
-                    .stream()
-                    .map(Result::getMetacard)
-                    .collect(Collectors.toList());
-
-            if (metacards.size() < deleteRequest.getAttributeValues()
-                    .size()) {
-                throw new StopProcessingException(
-                        "Unable to remove all metacards contained in request.");
-            }
-
-            return metacards;
         } catch (FederationException e) {
             LOGGER.debug("Unable to complete query for updated metacards.", e);
             throw new IngestException("Exception during runtime while performing delete");
         }
+
+        List<Metacard> metacards = query.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .collect(Collectors.toList());
+
+        if (notFoundValidMetacards(deleteRequest, query)) {
+            logFailedQueryInfo(deleteRequest, query);
+            throw new StopProcessingException("Unable to remove all metacards contained in request.");
```

Comment:
Should this message say something similar to what you have in your log statements?